src/accelerate/data_loader.py

```diff
@@ -495,10 +495,6 @@ def __init__(self, dataset, split_batches: bool = False, skip_batches=0, _drop_l
         self.state = AcceleratorState()
         self._drop_last = _drop_last
         self.skip_batches = skip_batches
-        # We can safely pass because the default is -1
-        with suppress(Exception):
-            length = getattr(self.dataset, "total_dataset_length", len(self.dataset))
-            self.remainder = length % self.total_batch_size
```

Comment:
Is there a reason this has been moved into `__iter__` and not `__init__`?

Reply:
I didn't go through the rest of this repo to figure out how ``remainder`` is used. However, since ``self.reset()`` will always set it to -1, it just makes more sense (looking at data_loader.py) to follow ``DataLoaderShard`` and set the remainder in ``__iter__``. Otherwise, it should be safe to remove this code completely.

Reply:
It's used for Gradient accumulation. Here it makes sense to have it in `__iter__` as there's no need to calculate it twice

Reply:
Agreed

Reply:
Well, I might be missing something, but, in ``__iter__``, ``self.reset()`` will set ``self.remainder`` to -1. So, ``self.remainder`` won't be useful once we start iterating through the dataloader/dataset. Is that correct?

Reply:
Looking more, it is also updated at the end of the iter loop. @muellerzr do we actually need those lines?

But while we investigate, I agree that it's safer to just copy this after the reset.

Reply:
Hmmm... will look into this today.